### ISSUE TITLE
Moving InputfieldPageAutocomplete's "delete item" click handler into init method?

### ISSUE THREAD

**Issue by apeisa** [ryancramerdesign/ProcessWire#1183]
For some reason click handler for removing rows is initialized on document ready instead of init method. In normal pw admin use this doesn't matter, but we are building product where we are loading inputfields through ajax. Moving handler from row https://github.com/ryancramerdesign/ProcessWire/blob/master/wire/modules/Inputfield/InputfieldPageAutocomplete/InputfieldPageAutocomplete.js#L247 inside init method in same file naturally fixes the problem and should be fully backwards compatible solution. $(document).ready(function() wouldn't be necessary at all.

Any hope to get that moved?

**Comment by ryancramerdesign** [comment]
I think we could do that, though not 100% sure it would be fully backwards compatible. I look at it and think "why is that in document.ready rather than in the dedicated class?" When I wrote the code there, it's always possible I was just being dumb, but it's also possible there was a good reason. So before we move it, I'd want to see if there's some way to make it work with what you are trying to do there. For instance, what if we changed this:
`````
$(".InputfieldPageAutocomplete ol").on('click', 'a.itemRemove', function() {
`````
To this?
`````
$(document).on('click', '.InputfieldPageAutocomplete a.itemRemove', function() {
``````
